### PR TITLE
fix that submit draft keeps on spinning

### PR DESCRIPTION
**Pull request by petersalomonsen**
Based on the report that some user only saw the submit button spinner keeping on spinning, which I assume is as reproduced in the test that can be seen in the video here:

https://github.com/user-attachments/assets/04d0e6f2-9d65-458e-95cd-729bce1173a4

The test reproduces a scenario where the transaction completes, but the view call for getting all proposal ids does not show the latest proposal immediately, which might happen in the case the transaction is not fully finalized. The previous implementation will then just keep on spinning, and so it is assumed that this is what the user experienced.

By retrying checking for the latest proposal until it matches the submitted title and description, the submitted proposal will be found eventually, and the submit button will stop spinning, and the new proposal should display.

With the fix, the submit button stops spinning, and the submitted proposal is displayed:

https://github.com/user-attachments/assets/8eef24db-f08b-450f-8b52-548a133af94c

### CODE REVIEW

**Comment by petersalomonsen** [comment]
@race-of-sloths include

**Comment by trechriron** [comment]
Love this!

**Review by Megha-Dev-19 — APPROVED** [review]
LGTM
I think we need to make this change to RFP as well in infra.

**Comment by Megha-Dev-19** [comment]
@race-of-sloths score 3

**Comment by race-of-sloths** [comment]
@petersalomonsen Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
King of Sloths shakes your hand! Yellow leader's jersey is yours deservedly!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&pr=NEAR-DevHub/neardevhub-bos/951&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&pr=NEAR-DevHub/neardevhub-bos/951&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/petersalomonsen)


<details><summary>Current status: <i>executed</i></summary>

| Reviewer | Score |
|--------|--------|
| @Megha-Dev-19  | 3 |


Your contribution is much appreciated with a final score of 3!
You have received 53 (30 base + 20 weekly bonus + 5% lifetime bonus) Sloth points for this contribution


Another weekly streak completed, well done @petersalomonsen! To keep your weekly streak and get another bonus make pull request next week! Looking forward to see you in race-of-sloths
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/petersalomonsen)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Pull request merged**
